src/CosmosEditorManager.ts

```diff
@@ -90,12 +88,12 @@ export class CosmosEditorManager implements vscode.Disposable {
             const dontShow: boolean | undefined = globalState.get(this.dontShowKey);
             if (dontShow !== true) {
                 const message: string = `Saving 'cosmos-editor.json' will update the entity "${editor.label}" to the Cloud.`;
-                const result: string | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.OK, DialogBoxResponses.DontShowAgain);
+                const result: MessageItem | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.OK, DialogBoxResponses.DontShowAgain, DialogBoxResponses.Cancel);
 
-                if (!result) {
-                    throw new UserCancelledError();
-                } else if (result === DialogBoxResponses.DontShowAgain) {
+                if (result === DialogBoxResponses.DontShowAgain) {
                     await globalState.update(this.dontShowKey, true);
```

Comment:
Shouldn't this throw canceled?

Reply:
In this case "Don't show again" means "Upload and don't warn again". We've actually been discussing improving that message and I just filed a bug here: https://github.com/Microsoft/vscode-cosmosdb/issues/280

For now, this PR keeps the behavior the same as it was and I would prefer to fix the message later (since people have strong opinions on that and not this change)

Reply:
Yep, I had assumed it meant "no and don't ask again".